### ISSUE TITLE
Gulp Sass Error with Yarn

### ISSUE THREAD

**Issue by manelmadeira** [dlmanning/gulp-sass#563]
After running `yarn`, i'm having this issue with a dependency, node-sass.

```return binding.readdir(pathModule._makeLong(path), options.encoding);
           

Error: ENOENT: no such file or directory, scandir '/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/node-sass/vendor'
    at Object.fs.readdirSync (fs.js:862:18)
    at Object.getInstalledBinaries (/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/node-sass/lib/extensions.js:121:13)
    at foundBinariesList (/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/node-sass/lib/errors.js:20:15)
    at foundBinaries (/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/node-sass/lib/errors.js:15:5)
    at Object.module.exports.missingBinary (/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/node-sass/lib/errors.js:45:5)
    at module.exports (/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/node-sass/lib/binding.js:15:30)
    at Object.<anonymous> (/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/node-sass/lib/index.js:14:35)
    at Module._compile (module.js:573:32)
    at Object.Module._extensions..js (module.js:582:10)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)
    at Function.Module._load (module.js:441:3)
    at Function.Module._load (/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/gulp-hub/lib/load-subfile.js:26:32)
    at Module.require (module.js:500:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/manuelmadeira/Documents/cashyt/cashyt-dashboard/node_modules/gulp-sass/index.js:187:21)
```

They fixed the issue on a new version. 

Can you help, please?

Thanks,
Manuel

**Comment by Danita** [comment]
Try

```
nodejs node_modules/node-sass/scripts/install.js
npm rebuild node-sass
```

**Comment by manelmadeira** [comment]
@Danita it worked, but it's a temporary solution. 

We need that fixed.

**Comment by rossb** [comment]
+1 for a fix on this.

I've been looking back through the changelog, but not sure what version of gulp-sass I should downgrade to in the meantime to avoid this error?

I can't use the fix above to resolve, as I need to be able to share my repo without a footnote to manually build (worth noting -- I get the same error using either `yarn` or `npm install`).

**Comment by rossb** [comment]
After trying some older versions I can see downgrading doesn't appear to fix this issue. Is there any way I can side-step the error without either manually running install.js, or waiting for an update to `gulp-sass`? Should the issue only affect `yarn` but not `npm install`?

**Comment by xzyfer** [comment]
This is not a gulp-sass bug, it's a yarn bug. I've heard yarn@0.6.1 works correct, but a bug was introduced on 0.7.

**Comment by rossb** [comment]
Ah, I can see a fresh repo download then `npm install` does actually work in my case (as opposed to `yarn install`), sorry if I added confusion with my comments above. Previously I simply deleted node_modules then ran `npm install` after previously using yarn, which threw the same error for some reason (I guess cache related).

**Comment by xzyfer** [comment]
It looks like a bug with the yarn cache. It appears to have been fixed in v0.17.7.

**Comment by bf** [comment]
Maybe this is an issue because we do `npm install --save-dev gulp-sass` instead of normal `npm install --save gulp-sass`? Maybe it is just not running the install script?
